openstack_flavor_manager/cloud.py

```diff
@@ -28,8 +16,9 @@ def get_spec_or_default(key_string: str, flavor_spec: dict, defaults: dict, is_b
     else:
         raise ValueError(f"Unknown key_string '{key_string}'")
 
-    if is_bool:
-        return str_to_bool(value)
+    if is_bool and not isinstance(value, bool):
```

Comment:
I dont think we need the ``is_bool`` at all anymore.

The type should be present in defaults and flavor spec dictionaries, and if it does not match the expected type the error will be (automatically) raised when the openstack api is called.